### PR TITLE
Yifanzhou/add migration guidance

### PR DESCRIPTION
**Pull request by yifan-zhou922**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Now in `azure-analytics-purview-datamap`, users should create a DataMapClient first and then build the client for each operation groups.
+
+```java
+ClientSecretCredential cred = new ClientSecretCredentialBuilder()
+    .tenantId(Configuration.getGlobalConfiguration().get("TENANT_ID"))
+    .authorityHost(Configuration.getGlobalConfiguration().get("AUTHORITY_HOST"))
+    .clientId(Configuration.getGlobalConfiguration().get("CLIENT_ID"))
+    .clientSecret(Configuration.getGlobalConfiguration().get("CLIENT_SECRET"))
+    .build();
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md:42]: I think the credential has nothing to do with migration.

File: sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+- [General changes](#general-changes)
+  - [Package and client name](#package-and-client-name)
+- [Additional samples](#additional-samples)
+
+## Migration benefits
+
+> Note: `azure-analytics-purview-catalog` has been [deprecated]. Please upgrade to `azure-analytics-purview-datamap` for continued support.
+
+
+The new Purview DataMap library `azure-analytics-purview-datamap` includes the service models together with the GA DataMap APIs [API Document](https://learn.microsoft.com/rest/api/purview/datamapdataplane/operation-groups?view=rest-purview-datamapdataplane-2023-09-01). The client name and the operation names have slightly changed but the main functionality remains the same.
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md:19]: nit, we should use full word, instead of "GA"

Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md]: Updated. Thanks.

File: sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md (hunk 1 above)
Comment (yifan-zhou922) [reply sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md]: Removed.

File: sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    .endpoint(System.getenv("<account-name>.purview.azure.com"))
+    .credential(new DefaultAzureCredentialBuilder().build())
+    .buildClient();
+```
+
+Now in `azure-analytics-purview-datamap`, users should create a DataMapClient first and then build the client for each operation groups.
+
+```java
+DataMapClientBuilder clientBuilder = new DataMapClientBuilder()
+    .endpoint(Configuration.getGlobalConfiguration().get("ENDPOINT"))
```
Comment (weidongxu-microsoft) [inline sdk/purview/azure-analytics-purview-datamap/MigrationGuide.md:38]: I would suggest write the `.endpoint()` part same. I assume migration does not affect this part of the code.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**